api/internal/localizer/locloader.go

```diff
@@ -117,7 +117,8 @@ func (ll *locLoader) Load(path string) ([]byte, error) {
 // Otherwise, New returns an error.
 func (ll *locLoader) New(path string) (ifc.Loader, error) {
 	repoSpec, err := git.NewRepoSpecFromURL(path)
-	if err == nil && repoSpec.Ref == "" {
+	isRemote := err == nil
+	if isRemote && repoSpec.Ref == "" {
```

Comment:
The conditions here seem to be in the following order:

1. isRemote and repoSpec.Ref is empty
2. !isRemote
3. isRemote and repoSpec.Ref is not empty

This seems like kind of a weird order to put the code (separating the isRemote cases). I suggest nesting the if statements here, so it's more like

```
if isRemote {
  if repoSpec.Ref == "" {
  ...
  }
  else {
  ...
  }
} else {
  ... 
}
```

I think that would be easier to read the if branches.

Reply:
Oh, I looked a bit closer and see why it is structured the way it is. Maybe Yuwen's suggestion above to run the not remote code first would be a better way to structure it.

Reply:
Yuwen's suggestion above to do the not remote case first may also help with readability here.

Reply:
Hope it's easier to read now.